### PR TITLE
Num method bspde

### PR DESCRIPTION
**Pull request by animoku168**
This is current changes to the next presentation

### CODE REVIEW

File: src/numerical_method.py [hunk 1]
```diff
@@ -171,8 +171,7 @@ def __init__(self) -> None:
 
 
 class PDEParams(Params):
-    def __init__(self, und_step: int, time_step: int, stock_min: int, stock_max: int, method: str) -> None:
-        # todo: to be implemented
+    def __init__(self, und_step: int = 2, time_step: float = 1/1200, stock_min: int = 0, stock_max: int = 200, method: str = 'explicit') -> None:
```
Comment (friedgab) [inline src/numerical_method.py:174]: Use enum for method.

File: test/test_pricer.py [hunk 2]
```diff
@@ -139,3 +139,18 @@ def test_analytic_pricer(self, underlying, ref_strike, expiry, model, greek_meth
         result_put_call = {key: result['call'].get(key, 0) + result['put'].get(key, 0)
                            for key in set(result['call']) | set(result['put'])}
         assert result_put_call == pytest.approx(result['fwd'], abs=1e-6)
+
+class TestEuropeanPDEPricer:
+    und = Stock.TEST_COMPANY
+    expiry = 2.0
+    strike = 0.95 * MarketData.get_spot()[und]
+    model = FlatVolModel(und)
+    params = [PDEParams(method="explicit"), PDEParams(method="implicit"), PDEParams(method="cranknicolson")]
```
Comment (friedgab) [inline test/test_pricer.py:148]: Use enum

File: src/numerical_method.py (hunk 1 above)
Comment (friedgab) [reply src/numerical_method.py:174]: Is stock_max the edge of the grid? So is S within [0,200] ? We have a few underlying (MONEY_MALL) that has current spot >200. Shouldn't we use a multiplicator istead?

Comment (animoku168) [reply src/numerical_method.py]: This has been fixed now.

File: test/test_pricer.py (hunk 2 above)
Comment (animoku168) [reply test/test_pricer.py]: This has also been implemented

**Comment by animoku168** [comment]
Thanks to Gabor F for the suggestions to improve the robustness of the code

**Review by friedgab — APPROVED** [review]

**Pull request merged**